streams/src/test/java/org/apache/kafka/streams/integration/CleanUpIntegrationTest.java

```diff
@@ -0,0 +1,250 @@
+                                                            return new KeyValue<>(key, value);
+                                                        }
+                                                    })
+                                                    .groupByKey()
+                                                    .count("global-count");
+        globalCounts.to(Serdes.Long(), Serdes.Long(), OUTPUT_TOPIC);
+
+        final KStream<Long, Long> windowedCounts = input
+                                                       .through(INTERMEDIATE_USER_TOPIC)
+                                                       .map(new KeyValueMapper<Long, String, KeyValue<Long, String>>() {
+                                                           @Override
+                                                           public KeyValue<Long, String> apply(final Long key, final String value) {
+                                                               Utils.sleep(1000);
+                                                               return new KeyValue<>(key, value);
+                                                           }
+                                                       })
+                                                       .groupByKey()
+                                                       .count(TimeWindows.of(35).advanceBy(10), "count")
+                                                       .toStream()
+                                                       .map(new KeyValueMapper<Windowed<Long>, Long, KeyValue<Long, Long>>() {
+                                                           @Override
+                                                           public KeyValue<Long, Long> apply(final Windowed<Long> key, final Long value) {
+                                                               return new KeyValue<>(key.window().start() + key.window().end(), value);
+                                                           }
+                                                       });
+        windowedCounts.to(Serdes.Long(), Serdes.Long(), OUTPUT_TOPIC_2);
+
+        return builder;
+    }
+
+    private void cleanGlobal() {
+        final Properties cleanUpConfig = new Properties();
+        cleanUpConfig.put(ConsumerConfig.HEARTBEAT_INTERVAL_MS_CONFIG, 100);
+        cleanUpConfig.put(ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG, "" + CONSUMER_TIMEOUT);
+
+        final int rc = new StreamsCleanupClient().run(
+            new String[]{
+                "--application-id", APP_ID,
+                "--bootstrap-server", CLUSTER.bootstrapServers(),
+                "--zookeeper", CLUSTER.zKConnectString(),
+                "--source-topics", INPUT_TOPIC,
+                "--intermediate-topics", INTERMEDIATE_USER_TOPIC
+            },
+            cleanUpConfig);
+        Assert.assertEquals(0, rc);
+
+        // Check if all internal topics got deleted
```

Comment:
Ah, this comment confused me for a while.

Perhaps rename `expectedTopics` to sth like `expectedRemainingTopicsAfterCleanup`?